src/bp-core/classes/class-bp-embed.php

```diff
@@ -341,6 +341,11 @@ public function autoembed( $content, $type = '' ) {
 			}
 		}
 
+		// Check if blog or CPT post activity has link/video embed.
+		if ( is_object( $type ) && ! empty( $type->component ) && 'blogs' === $type->component ) {
+			$link_embed = true;
+		}
```

Comment:
@surajkrsingh 
The `$link_embed` variable is set to `true` even when a blog post doesn't contain any embeds. Please investigate and fix the condition that determines the value of `$link_embed`